### PR TITLE
Rav 4558 Nulls First And Nulls Last Support In MSSql

### PR DESCRIPTION
**Pull request by roy-sounak**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java [hunk 1]
```diff
@@ -28,39 +29,73 @@
...
-  @Override public void unparseDateTimeLiteral(SqlWriter writer,
-      SqlAbstractDateTimeLiteral literal, int leftPrec, int rightPrec) {
+  @Override protected SqlNode emulateNullDirectionWithIsNull(
+          SqlNode node, boolean nullsFirst, boolean desc) {
+    // No need to emulate null when we are presented with default conditions
+    // like (Asc & Nulls First) or (Dsc & Nulls Last)
+    if (nullCollation.isDefaultOrder(nullsFirst, desc)) {
+      return null;
+    }
+    String sqlLiteralForCaseCall = (!(nullsFirst && desc)) ? "1" : "0";
```
Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:80]: Please don't use negation here

Comment (soma-mondal) [inline core/src/main/java/org/apache/calcite/sql/dialect/MssqlSqlDialect.java:80]: Please rename sqlLiteralForCaseCall and define else part here as well.

**Review by soma-mondal — APPROVED** [review]

**Review by soma-mondal — CHANGES_REQUESTED** [review]
Please revert HiveSqlDialect changes

**Comment by soma-mondal** [comment]
L1 Done

**Review by soma-mondal — APPROVED** [review]

**Review by yogeshmuley — APPROVED** [review]

**Review by yogeshmuley — APPROVED** [review]

**Pull request merged**